packages/algolia/src/index.ts

```diff
@@ -1,23 +1,32 @@
 import algoliasearch, { SearchClient } from 'algoliasearch';
 import { AlgoliaSearchClient, Apps } from './client';
 import * as gp2 from './gp2';
+import { NoTokenAlgoliaClient } from './no-token-client';
 
 export type { SearchResponse } from '@algolia/client-search';
 export * from './client';
 export * from './crn/types';
 export * from './filters';
 export { gp2 };
 
+export type AlgoliaClient<App extends Apps> =
+  | AlgoliaSearchClient<App>
+  | NoTokenAlgoliaClient<App>;
+
 type AlgoliaSearchClientNativeFactoryParams = {
-  algoliaApiKey: string;
+  algoliaApiKey: string | null;
   algoliaAppId: string;
 };
 
 export const algoliaSearchClientNativeFactory = ({
```

Comment:
i dont think this is used anywhere outside of BE and i guess there is no point returning a client without a key, maybe throw an error instead?

Reply:
Oki